src/charm.py

```diff
@@ -162,6 +177,18 @@ def _push_schemas(self):
                 schema = f.read()
             self._container.push(Path(self._config_dir_path, schema_file), schema, make_dirs=True)
 
+    def _get_hydra_endpoint_info(self) -> str:
+        # Assign a default value as config file won't accept an empty string
+        oauth2_provider_url = "http://127.0.0.1:4445/"
```

Comment:
This default does not make sense in the context of k8s. I think it would be better to just return an `None` oauth2_provider_url and handle it in the config template.

If `oauth2_provider_url` is not set, then the template should not add the `oauth2_provider_url` the config file, e.g.:
```
{%- if oauth2_provider_url %}
oauth2_provider:
    url: {{ oauth2_provider_url }}
{%- endif %}
```